substrate/frame/identity/src/lib.rs

```diff
@@ -688,16 +664,14 @@ pub mod pallet {
 		///
 		/// - `index`: the index of the registrar whose fee is to be set.
 		/// - `fields`: the fields that the registrar concerns themselves with.
-		///
-		/// ## Complexity
-		/// - `O(R)`.
-		///   - where `R` registrar-count (governance-bounded).
 		#[pallet::call_index(8)]
-		#[pallet::weight(T::WeightInfo::set_fields(T::MaxRegistrars::get()))] // R
+		#[pallet::weight(T::WeightInfo::set_fields(T::MaxRegistrars::get()))]
 		pub fn set_fields(
 			origin: OriginFor<T>,
 			#[pallet::compact] index: RegistrarIndex,
-			fields: IdentityFields,
+			fields: IdentityFields<
```

Comment:
Why isn't this just a `BitVec` of length len(identity_fields).

Reply:
What I meant there is that `IdentityFields` should have been some associated type on `IdentityInfo`. The on chain logic doesn't care what this type is.

Reply:
Yeah, but this wrapper can just go into the `legacy` module as well.